react/src/components/MaintenanceSettingList.tsx

```diff
@@ -0,0 +1,150 @@
+import { useSuspendedBackendaiClient } from '../hooks';
+import { useSetBAINotification } from '../hooks/useBAINotification';
+import { usePainKiller } from '../hooks/usePainKiller';
+import SettingList, { SettingGroup } from './SettingList';
+import { RedoOutlined } from '@ant-design/icons';
+import { Button } from 'antd';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+const MaintenanceSettingList = () => {
+  const [isRecalculating, setIsRecalculating] = useState(false);
+  const [isRescanning, setIsRescanning] = useState(false);
+
+  const { t } = useTranslation();
+  const { upsertNotification } = useSetBAINotification();
+  const baiClient = useSuspendedBackendaiClient();
+  const painKiller = usePainKiller();
+
+  const recalculateUsage = () => {
+    setIsRecalculating(true);
+
+    upsertNotification({
+      message: t('maintenance.RecalculateUsage'),
+      description: t('maintenance.Recalculating'),
+      open: true,
+      backgroundTask: {
+        status: 'pending',
+        promise: baiClient.maintenance
+          .recalculate_usage()
+          .finally(() => setIsRecalculating(false)),
+        statusDescriptions: {
+          pending: t('maintenance.Recalculating'),
+          resolved: t('maintenance.RecalculationFinished'),
+          rejected: t('maintenance.RecalculationFailed'),
+        },
+      },
+    });
+  };
+
+  const rescanImages = async () => {
```

Comment:
Since rescan_images() displays the error via ok and msg, it doesn't seem to need to be determined by a try_catch statement.

```suggestion
  const rescanImages = async () => {
    setIsRescanning(true);

    const { rescan_images } = await baiClient.maintenance
      .rescan_images()
      .finally(() => setIsRescanning(false));

    if (!rescan_images.ok) {
      upsertNotification({
        message: t('maintenance.RescanImages'),
        description: rescan_images.msg ?? t('maintenance.RescanFailed'),
      });
      return;
    }

    upsertNotification({
      key: rescan_images.task_id,
      message: t('maintenance.RescanImages'),
      open: true,
      duration: 0,
      backgroundTask: {
        status: 'pending',
        percent: 0,
        taskId: rescan_images.task_id,
        statusDescriptions: {
          pending: t('maintenance.RescanImageScanning'),
          resolved: t('maintenance.RescanImageFinished'),
          rejected: t('maintenance.RescanFailed'),
        },
      },
    });
  };
```